source/game/field/ObjectDirector.cc

```diff
@@ -42,10 +48,70 @@ void ObjectDirector::addObject(ObjectCollidable *obj) {
     m_objects.push_back(obj);
 }
 
+/// @addr{0x8082AB04}
+u32 ObjectDirector::checkKartObjectCollision(Kart::KartObject *kartObj,
+        ObjectCollisionConvexHull *convexHull) {
+    u32 count = 0;
+    ObjectCollidable *obj;
+
+    while ((obj = reinterpret_cast<ObjectCollidable *>(
+                    BoxColManager::Instance()->getNextObject()))) {
+        auto *objCollision = obj->collision();
+        if (!objCollision) {
+            continue;
+        }
+
+        obj->calcCollisionTransform();
+        if (!obj->checkCollision(convexHull, m_hitDepths[count])) {
+            continue;
+        }
+
+        // We have a collision, process it
+        // Assume that we are not in a star, mega, or bullet
+        Kart::Reaction reactionOnKart = static_cast<Kart::Reaction>(
+                m_hitTableKart.reaction(m_hitTableKart.slot(obj->id())));
```

Comment:
```suggestion
        Kart::Reaction reactionOnKart = m_hitTableKart.reaction(m_hitTableKart.slot(obj->id()));
```
The static cast should be done in the `reaction` function itself.